runtime/src/qio/qio_popen.c

```diff
@@ -33,9 +33,46 @@
 #include <unistd.h>
 #include <spawn.h>
 
+// We need to be able to call malloc, free, etc.
+#include "chpl-mem-no-warning-macros.h"
+
 // current environment variables
 extern char** environ;
 
+/* Helper routines for allocating out of the C
+   heap instead of the Chapel one. That is necessary
+   since the Chapel heap might not be available to
+   a forked process, and we need args for exec
+   (command, arguments, environment) to be available
+   after a fork.
+ */
```

Comment:
Please include a little more information here about the circumstances in which the heap isn't reachable (gasnet with seg=fast or large on Cray X* systems, thus hugepages for the heap; and does this also happen for comm=ugni on Cray X*, or does it work there? it would be useful to mention that, either way). This will be useful if someone ever tries to solve this problem in another way, but more likely, it could be useful if someone else runs into a similar problem in the future.